src/EFCore.Analyzers/UninitializedDbSetDiagnosticSuppressor.cs

```diff
@@ -0,0 +1,90 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.EntityFrameworkCore;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public class UninitializedDbSetDiagnosticSuppressor : DiagnosticSuppressor
+{
+    private static readonly SuppressionDescriptor _suppressUninitializedDbSetRule = new(
+        id: "SPR1001",
+        suppressedDiagnosticId: "CS8618",
+        justification: "DbSet properties on DbContext subclasses are automatically populated by the DbContext constructor");
+
+    /// <inheritdoc />
+    public override ImmutableArray<SuppressionDescriptor> SupportedSuppressions => ImmutableArray.Create(_suppressUninitializedDbSetRule);
```

Comment:
Longer syntax, but avoids multiple allocations if the property is accessed multiple times.

```suggestion
    public override ImmutableArray<SuppressionDescriptor> SupportedSuppressions { get; } = ImmutableArray.Create(_suppressUninitializedDbSetRule);
```

Reply:
Makes sense, thanks.